### PR TITLE
Refactor component for new helm chart

### PR DESCRIPTION
**Pull request by TheBigLee**
This PR updates the component to use the new upstream helm chart

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,6 +51,33 @@ docs-serve: ## Preview the documentation
 .PHONY: test
 test: commodore_args += -f tests/$(instance).yml
 test: .compile ## Compile the component
+.PHONY: gen-golden
```
Comment (zugao) [inline Makefile]:
```suggestion

.PHONY: gen-golden
```

File: class/defaults.yml [hunk 2]
```diff
@@ -4,18 +4,18 @@ parameters:
...
     charts:
-      minio: 8.0.0
+      minio:
+        source: https://charts.min.io
+        version: 5.0.13
     adminCredentials:
       secretName: admin-credentials-${_instance}
-      accessKey: ?{vaultkv:${cluster:tenant}/${cluster:name}/minio/${_instance}-accesskey}
-      secretKey: ?{vaultkv:${cluster:tenant}/${cluster:name}/minio/${_instance}-secretkey}
+      rootUser: ?{vaultkv:${cluster:tenant}/${cluster:name}/minio/${_instance}-accesskey}
```
Comment (zugao) [inline class/defaults.yml:12]: don't forget to add the entries for each cluster.

**Review by zugao — APPROVED** [review]
LGTM, some small things

**Comment by TheBigLee** [comment]
Just realized that I should add a migration documentation as well. Will do that later today.

File: class/defaults.yml (hunk 2 above)
Comment (TheBigLee) [reply class/defaults.yml:12]: What do you mean?

Comment (zugao) [reply class/defaults.yml:12]: In vault :)

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 3]
```diff
@@ -17,25 +17,41 @@ default:: `syn-minio`
...
 type:: string
 default:: Vault reference
 
-A vault reference pointing to the Vault secret containing the access key.
-If the key isn't provided as a vault reference, the secret key must be provided in its base64 encoded form.
-It can be generated using the command `echo -n '<mySecretKey> | base64`.
+A vault reference pointing to the Vault secret containing the rootUser or a string with the username for the admin user
+
+NOTE: Deprecated: Use `adminCredentials.rootPassword` parameter instead.
+
```
Comment (simu) [inline docs/modules/ROOT/pages/references/parameters.adoc]: Personal taste: I would completely remove the documentation for the deprecated parameters

**Review by simu — APPROVED** [review]
Implementation LGTM, I didn't verify the golden test YAML

**Review by zugao — APPROVED** [review]
LGTM

File: docs/modules/ROOT/pages/references/parameters.adoc (hunk 3 above)
Comment (TheBigLee) [reply docs/modules/ROOT/pages/references/parameters.adoc]: yeah makes sense. Removed them

**Comment by TheBigLee** [comment]
**WARNING**
Just found out one huge issue, that needs to be addressed before merging this:

This upgrade will change the minio server setup from deployment to statefulset. Meaning that the PVCs will not be re-used.
This will therefore most likely result in a complete data loss, as the PVCs will be removed when the deployment is removed and replaced by a statefulset

**Pull request closed without merging**